server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -637,10 +637,10 @@ public void reset(CopyTo copyTo) {
         private final TriFunction<String, MappingParserContext, Object, T> parser;
         private final Function<FieldMapper, T> initializer;
         private boolean acceptsNull = false;
-        private List<Consumer<T>> validators = new ArrayList<>();
-        private Serializer<T> serializer = XContentBuilder::field;
```

Comment:
So does this always pick `XContentBuilder#field(String, Object)` then? Does the `<T>` big get erased somewhere?

Reply:
Certainly looks that way from profiling + a quick experiment using the debugger:

![image](https://user-images.githubusercontent.com/6490959/150140694-91e87eac-612e-4fa8-acc6-fe9f92368959.png)

Also in some cases the type structure is so messy that the most optimal method isn't the `#field` and `#stringListField` (in this case we simply can't add that kind of overload because of other existing overloads that conflict) is the correct choice for performance.

Reply:
Profiling (excerpt, but in 100% of paths I see the unknown value used, even though we have all kinds of parameters in our benchmark mappings):

<img width="2150" alt="image" src="https://user-images.githubusercontent.com/6490959/150141928-ffdd5c7c-0b17-43b6-a975-fc2e02e5174a.png">

Reply:
Do we still get the efficient version if we create a constructor that then passes `XContentBuilder::field, Object::toString` as default parameters to `this`.  Just wondering if there's a way to avoid adding two new essentially identical parameters to twenty callsites...

Reply:
No that doesn't work. I tried it yesterday already. As far as I'm aware the compiler can't generate multiple overloads from a single constructor method for generics like that (which I think also explains why we always link to the object version).

Reply:
Schade.  OK, can you add a comment to the constructor explaining why we need to always pass this stuff in?

Reply:
Schade indeed :) I tried to find a reasonable comment for that now. It was a little tricky to come up with one that captures this discussion, let me know what you think.